In [ ]:
import pandas as pd

In [ ]:
a = [1,2,3,4,5,6,7]
pd.Series(a)

0    1
1    2
2    3
3    4
4    5
5    6
6    7
dtype: int64

In [ ]:
b = {"banana" :2, "apple" : 3, "oranges":2, "carrots" : 5, "lime" : 1}
pd.Series(b)

banana     2
apple      3
oranges    2
carrots    5
lime       1
dtype: int64

In [ ]:
b = [["banana", 2, "dog"] , ["oranges", 3, "cat"], ["apples", 4, "dear"] ]
pd.DataFrame(b)

0  1     2
0   banana  2   dog
1  oranges  3   cat
2   apples  4  dear

In [ ]:
b = [["banana", 2, "dog"] , ["oranges", 3, "cat"], ["apples", 4, "dear"] ]
pd.DataFrame(b, columns=['fruits', 'numbers', 'animal'], index = [1,2,3])

fruits  numbers animal
1   banana        2    dog
2  oranges        3    cat
3   apples        4   dear

In [ ]:
data = {"Redg no" : [106, 107, 108, 109, 110], "Marks": [100, 150, 200, 250, 300], "percentage" : [20, 30, 40,15, 50]}
data2 = pd.DataFrame(data) #Data frame consists of series

In [ ]:
test_file = pd.read_csv("/content/diabetic_dataupdated.csv")

In [ ]:
file = test_file.to_csv("againtesting.csv")

In [ ]:
scores = pd.read_csv("/content/exam_scores.csv")

In [ ]:
scores.head()

gender race/ethnicity parental level of education         lunch  \
0    male        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2    male        group C                some college  free/reduced   
3  female        group D           bachelor's degree  free/reduced   
4    male        group D                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          74             68             67  
1               completed          58             68             66  
2                    none          66             65             65  
3                    none          74             75             73  
4                    none          78             77             71

In [ ]:
scores.tail()

gender race/ethnicity parental level of education         lunch  \
995  female        group C            some high school      standard   
996  female        group E                some college      standard   
997  female        group E          associate's degree  free/reduced   
998  female        group C                 high school      standard   
999    male        group C                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
995                    none          68             77             72  
996                    none          98             81             94  
997                    none          67             67             67  
998                    none          63             68             70  
999                    none          49             57             50

In [ ]:
shape = scores.shape
print(shape)

(1000, 8)


In [ ]:
scores.tail(100)

gender race/ethnicity parental level of education         lunch  \
900    male        group B            some high school  free/reduced   
901    male        group A          associate's degree      standard   
902    male        group E                some college  free/reduced   
903    male        group B            some high school      standard   
904    male        group E                some college  free/reduced   
..      ...            ...                         ...           ...   
995  female        group C            some high school      standard   
996  female        group E                some college      standard   
997  female        group E          associate's degree  free/reduced   
998  female        group C                 high school      standard   
999    male        group C                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
900                    none          55             59             54  
901                    none          87             78             78  
902                    none          73             63             60  
903               completed          83             82             78  
904                    none          81             76             74  
..                      ...         ...            ...            ...  
995                    none          68             77             72  
996                    none          98             81             94  
997                    none          67             67             67  
998                    none          63             68             70  
999                    none          49             57             50  

[100 rows x 8 columns]

In [ ]:
scores.dtypes

gender                         object
race/ethnicity                 object
parental level of education    object
lunch                          object
test preparation course        object
math score                      int64
reading score                   int64
writing score                   int64
dtype: object

In [ ]:
scores.info

<bound method DataFrame.info of      gender race/ethnicity parental level of education         lunch  \
0      male        group B           bachelor's degree      standard   
1    female        group C                some college      standard   
2      male        group C                some college  free/reduced   
3    female        group D           bachelor's degree  free/reduced   
4      male        group D                some college      standard   
..      ...            ...                         ...           ...   
995  female        group C            some high school      standard   
996  female        group E                some college      standard   
997  female        group E          associate's degree  free/reduced   
998  female        group C                 high school      standard   
999    male        group C                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
0                      none    

In [ ]:
scores.iloc[0,0]

'male'

In [ ]:
scores.iloc[0:4,2] #First 5 rows and 5th coloumn

0    bachelor's degree
1         some college
2         some college
3    bachelor's degree
Name: parental level of education, dtype: object

In [ ]:
scores.iloc[:,:]

gender race/ethnicity parental level of education         lunch  \
0      male        group B           bachelor's degree      standard   
1    female        group C                some college      standard   
2      male        group C                some college  free/reduced   
3    female        group D           bachelor's degree  free/reduced   
4      male        group D                some college      standard   
..      ...            ...                         ...           ...   
995  female        group C            some high school      standard   
996  female        group E                some college      standard   
997  female        group E          associate's degree  free/reduced   
998  female        group C                 high school      standard   
999    male        group C                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
0                      none          74             68             67  
1                 completed          58             68             66  
2                      none          66             65             65  
3                      none          74             75             73  
4                      none          78             77             71  
..                      ...         ...            ...            ...  
995                    none          68             77             72  
996                    none          98             81             94  
997                    none          67             67             67  
998                    none          63             68             70  
999                    none          49             57             50  

[1000 rows x 8 columns]

In [ ]:
scores.iloc[[0,1,2,3],2]

0    bachelor's degree
1         some college
2         some college
3    bachelor's degree
Name: parental level of education, dtype: object

In [ ]:
scores.loc[0:4, ["gender", "math score", "lunch"]]

gender  math score         lunch
0    male          74      standard
1  female          58      standard
2    male          66  free/reduced
3  female          74  free/reduced
4    male          78      standard

In [ ]:
scores["lunch"]

0          standard
1          standard
2      free/reduced
3      free/reduced
4          standard
           ...     
995        standard
996        standard
997    free/reduced
998        standard
999    free/reduced
Name: lunch, Length: 1000, dtype: object

In [ ]:
scores [["math score", "reading score"]]

math score  reading score
0            74             68
1            58             68
2            66             65
3            74             75
4            78             77
..          ...            ...
995          68             77
996          98             81
997          67             67
998          63             68
999          49             57

[1000 rows x 2 columns]

In [ ]:
scores.gender == "gay"

0      False
1      False
2      False
3      False
4      False
       ...  
995    False
996    False
997    False
998    False
999    False
Name: gender, Length: 1000, dtype: bool

In [ ]:
scores.lunch = "Biryani"

In [ ]:
scores.tail()



gender race/ethnicity parental level of education    lunch  \
995  female        group C            some high school  Biryani   
996  female        group E                some college  Biryani   
997  female        group E          associate's degree  Biryani   
998  female        group C                 high school  Biryani   
999    male        group C                some college  Biryani   

    test preparation course  math score  reading score  writing score  
995                    none          68             77             72  
996                    none          98             81             94  
997                    none          67             67             67  
998                    none          63             68             70  
999                    none          49             57             50

In [ ]:
scores.describe()

math score  reading score  writing score
count  1000.000000     1000.00000    1000.000000
mean     67.128000       70.17400      68.973000
std      14.815367       14.85599      15.109155
min      15.000000       18.00000      10.000000
25%      58.000000       60.00000      59.000000
50%      67.000000       70.00000      69.000000
75%      78.000000       81.00000      80.000000
max     100.000000      100.00000     100.000000

In [ ]:
scores.describe(include = 'object') #Only about catagorical column


gender race/ethnicity parental level of education    lunch  \
count     1000           1000                        1000     1000   
unique       2              5                           6        1   
top     female        group C                some college  Biryani   
freq       502            294                         226     1000   

       test preparation course  
count                     1000  
unique                       2  
top                       none  
freq                       654

In [ ]:
scores.describe(include = 'all') #about all column

gender race/ethnicity parental level of education    lunch  \
count     1000           1000                        1000     1000   
unique       2              5                           6        1   
top     female        group C                some college  Biryani   
freq       502            294                         226     1000   
mean       NaN            NaN                         NaN      NaN   
std        NaN            NaN                         NaN      NaN   
min        NaN            NaN                         NaN      NaN   
25%        NaN            NaN                         NaN      NaN   
50%        NaN            NaN                         NaN      NaN   
75%        NaN            NaN                         NaN      NaN   
max        NaN            NaN                         NaN      NaN   

       test preparation course   math score  reading score  writing score  
count                     1000  1000.000000     1000.00000    1000.000000  
unique                       2          NaN            NaN            NaN  
top                       none          NaN            NaN            NaN  
freq                       654          NaN            NaN            NaN  
mean                       NaN    67.128000       70.17400      68.973000  
std                        NaN    14.815367       14.85599      15.109155  
min                        NaN    15.000000       18.00000      10.000000  
25%                        NaN    58.000000       60.00000      59.000000  
50%                        NaN    67.000000       70.00000      69.000000  
75%                        NaN    78.000000       81.00000      80.000000  
max                        NaN   100.000000      100.00000     100.000000

The above table includes a summary of both numerical and categorical columns. A categorical column cannot have minimum or maximum values or mean and median. So the entries in these fields for categorical columns are

count - the count of non-null entries in the particular column. For example, the gender column has 1000 non-null entries.
unique - the count of unique values in a column. Only for categorical columns. For example, the gender column has two unique values - male and female.
top - this is also for only categorical columns. This tells us which category occurs the maximum number of times. For example, in the gender column, 'female' appears the maximum number of times.
freq - this is again for categorical columns only. This tells you the number of occurrences of that column's top category. For example, 'female' in the gender column occurs 502 times.
mean - the mean value of the numerical column. For example, the mean math score is 67.128.
std - this is the standard deviation of the numerical column. This tells you about the variation in the data. Don't worry if you don't know about it.
min - the minimum value in the numerical column. For example, the minimum math score is 15.0
25% - the 25th percentile (or 1st quartile) value in the numerical column. For example, the 25th percentile value for math score is 58.0.
50% - the 50th percentile (or 2nd quartile or the median) value in the numerical column. For example, the median math score is 67.0.
75% - the 75th percentile (or 3rd quartile) value in the numerical column. For example, the 75th percentile value for math score is 78.0.
max - the maximum value in the numerical column. For example, the maximum math score is 100.0.
NaN values means that a particular summary value is unavailable for a particular column. For example, gender (a categorical column) does not have a mean or median value as these are the properties of a numerical column only.


In [ ]:
scores["math score"].describe()

count    1000.000000
mean       67.128000
std        14.815367
min        15.000000
25%        58.000000
50%        67.000000
75%        78.000000
max       100.000000
Name: math score, dtype: float64

In [ ]:
scores.gender.describe()

count       1000
unique         2
top       female
freq         502
Name: gender, dtype: object

In [ ]:
scores["math score"].mean()

67.128

In [ ]:
scores["math score"].unique()

array([ 74,  58,  66,  78,  75,  69,  54,  65,  87,  64,  44,  57,  40,
        52,  82,  49,  37,  61,  73,  86,  48,  60,  76,  55,  72,  36,
        83,  70,  80,  56,  91,  67,  53,  77,  81,  89,  51,  96,  88,
        63,  71,  68,  38,  79,  90,  47,  33, 100,  42,  45,  62,  46,
        31,  59,  84,  94,  85,  92,  50,  34,  39,  43,  95,  93,  30,
        28,  35,  25,  97,  99,  98,  18,  23,  15])

In [ ]:
scores["math score"].value_counts()

63    35
61    32
62    32
76    30
60    29
      ..
28     1
35     1
18     1
23     1
15     1
Name: math score, Length: 74, dtype: int64

In [ ]:
scores.sort_values(by = "math score")

gender race/ethnicity parental level of education    lunch  \
949    male        group C                 high school  Biryani   
854    male        group C                 high school  Biryani   
891  female        group C                 high school  Biryani   
349    male        group B            some high school  Biryani   
739  female        group E            some high school  Biryani   
..      ...            ...                         ...      ...   
565    male        group E           bachelor's degree  Biryani   
911    male        group E                some college  Biryani   
128    male        group A          associate's degree  Biryani   
91     male        group D          associate's degree  Biryani   
495    male        group C             master's degree  Biryani   

    test preparation course  math score  reading score  writing score  
949                    none          15             18             10  
854                    none          18             30             18  
891                    none          23             31             27  
349                    none          25             31             30  
739                    none          25             35             38  
..                      ...         ...            ...            ...  
565               completed         100            100            100  
911                    none         100             95             87  
128               completed         100             97             94  
91                     none         100             94             96  
495               completed         100             96            100  

[1000 rows x 8 columns]

In [ ]:
scores.sort_values(by = "math score", ascending = False).head()

gender race/ethnicity parental level of education    lunch  \
91    male        group D          associate's degree  Biryani   
776   male        group E          associate's degree  Biryani   
588   male        group D                some college  Biryani   
128   male        group A          associate's degree  Biryani   
565   male        group E           bachelor's degree  Biryani   

    test preparation course  math score  reading score  writing score  
91                     none         100             94             96  
776               completed         100            100            100  
588               completed         100             85             91  
128               completed         100             97             94  
565               completed         100            100            100

In [ ]:
scores.sort_values(by = "math score", ascending = False).head()[:5]

gender race/ethnicity parental level of education    lunch  \
91    male        group D          associate's degree  Biryani   
776   male        group E          associate's degree  Biryani   
588   male        group D                some college  Biryani   
128   male        group A          associate's degree  Biryani   
565   male        group E           bachelor's degree  Biryani   

    test preparation course  math score  reading score  writing score  
91                     none         100             94             96  
776               completed         100            100            100  
588               completed         100             85             91  
128               completed         100             97             94  
565               completed         100            100            100

In [ ]:
scores.rename(index = {0 : "zero", 1 : "one", 2 : "two"}).head()

gender race/ethnicity parental level of education    lunch  \
zero    male        group B           bachelor's degree  Biryani   
one   female        group C                some college  Biryani   
two     male        group C                some college  Biryani   
3     female        group D           bachelor's degree  Biryani   
4       male        group D                some college  Biryani   

     test preparation course  math score  reading score  writing score  
zero                    none          74             68             67  
one                completed          58             68             66  
two                     none          66             65             65  
3                       none          74             75             73  
4                       none          78             77             71

In [ ]:
scores.isna()

gender  race/ethnicity  parental level of education  lunch  \
0     False           False                        False  False   
1     False           False                        False  False   
2     False           False                        False  False   
3     False           False                        False  False   
4     False           False                        False  False   
..      ...             ...                          ...    ...   
995   False           False                        False  False   
996   False           False                        False  False   
997   False           False                        False  False   
998   False           False                        False  False   
999   False           False                        False  False   

     test preparation course  math score  reading score  writing score  
0                      False       False          False          False  
1                      False       False          False          False  
2                      False       False          False          False  
3                      False       False          False          False  
4                      False       False          False          False  
..                       ...         ...            ...            ...  
995                    False       False          False          False  
996                    False       False          False          False  
997                    False       False          False          False  
998                    False       False          False          False  
999                    False       False          False          False  

[1000 rows x 8 columns]

In [ ]:
scores.isnull()

gender  race/ethnicity  parental level of education  lunch  \
0     False           False                        False  False   
1     False           False                        False  False   
2     False           False                        False  False   
3     False           False                        False  False   
4     False           False                        False  False   
..      ...             ...                          ...    ...   
995   False           False                        False  False   
996   False           False                        False  False   
997   False           False                        False  False   
998   False           False                        False  False   
999   False           False                        False  False   

     test preparation course  math score  reading score  writing score  
0                      False       False          False          False  
1                      False       False          False          False  
2                      False       False          False          False  
3                      False       False          False          False  
4                      False       False          False          False  
..                       ...         ...            ...            ...  
995                    False       False          False          False  
996                    False       False          False          False  
997                    False       False          False          False  
998                    False       False          False          False  
999                    False       False          False          False  

[1000 rows x 8 columns]

In [ ]:
scores.isnull().sum()

gender                         0
race/ethnicity                 0
parental level of education    0
lunch                          0
test preparation course        0
math score                     0
reading score                  0
writing score                  0
dtype: int64

In [ ]:
scores.Names.fillna("Unknown")

AttributeError: ignored

In [ ]:
scores = pd.read_csv("/content/exam_scores.csv")
scores.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [ ]:
kidney = pd.read_csv("/content/ChronicKidneyDisease.csv")

In [ ]:
kidney.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 26 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              400 non-null    int64  
 1   age             391 non-null    float64
 2   bp              388 non-null    float64
 3   sg              353 non-null    float64
 4   al              354 non-null    float64
 5   su              351 non-null    float64
 6   rbc             248 non-null    object 
 7   pc              335 non-null    object 
 8   pcc             396 non-null    object 
 9   ba              396 non-null    object 
 10  bgr             356 non-null    float64
 11  bu              381 non-null    float64
 12  sc              383 non-null    float64
 13  sod             313 non-null    float64
 14  pot             312 non-null    float64
 15  hemo            348 non-null    float64
 16  pcv             330 non-null    object 
 17  wc              295 non-null    obj

In [ ]:
kidney.describe(include = 'all').head()

id         age          bp          sg          al          su  \
count   400.0  391.000000  388.000000  353.000000  354.000000  351.000000   
unique    NaN         NaN         NaN         NaN         NaN         NaN   
top       NaN         NaN         NaN         NaN         NaN         NaN   
freq      NaN         NaN         NaN         NaN         NaN         NaN   
mean    199.5   51.483376   76.469072    1.017408    1.016949    0.450142   

           rbc      pc         pcc          ba  ...  pcv    wc   rc  htn   dm  \
count      248     335         396         396  ...  330   295  270  398  398   
unique       2       2           2           2  ...   44    92   49    2    5   
top     normal  normal  notpresent  notpresent  ...   41  9800  5.2   no   no   
freq       201     259         354         374  ...   21    11   18  251  258   
mean       NaN     NaN         NaN         NaN  ...  NaN   NaN  NaN  NaN  NaN   

        cad appet   pe  ane classification  
count   398   399  399  399            400  
unique    3     2    2    2              3  
top      no  good   no   no            ckd  
freq    362   317  323  339            248  
mean    NaN   NaN  NaN  NaN            NaN  

[5 rows x 26 columns]

In [ ]:
kidney.sort_values(by = "age", ascending = False)

id   age    bp     sg   al   su     rbc      pc         pcc          ba  \
200  200  90.0  90.0  1.025  1.0  0.0     NaN  normal  notpresent  notpresent   
171  171  83.0  70.0  1.020  3.0  0.0  normal  normal  notpresent  notpresent   
39    39  82.0  80.0  1.010  2.0  2.0  normal     NaN  notpresent  notpresent   
160  160  81.0  60.0    NaN  NaN  NaN     NaN     NaN  notpresent  notpresent   
237  237  80.0  70.0  1.015  2.0  2.0     NaN  normal  notpresent  notpresent   
..   ...   ...   ...    ...  ...  ...     ...     ...         ...         ...   
117  117   NaN  70.0  1.020  0.0  0.0     NaN     NaN  notpresent  notpresent   
169  169   NaN  70.0  1.010  0.0  2.0     NaN  normal  notpresent  notpresent   
191  191   NaN  70.0  1.010  3.0  0.0  normal  normal  notpresent  notpresent   
203  203   NaN  90.0    NaN  NaN  NaN     NaN     NaN  notpresent  notpresent   
268  268   NaN  80.0    NaN  NaN  NaN     NaN     NaN  notpresent  notpresent   

     ...  pcv     wc   rc  htn   dm  cad appet   pe  ane classification  
200  ...   37   7900  3.9  yes  yes   no  good   no   no            ckd  
171  ...   26  12800  3.1  yes   no   no  poor   no  yes            ckd  
39   ...   40   9800  4.2  yes  yes   no  good   no   no            ckd  
160  ...   35   9400  2.4  yes  yes  yes  poor  yes   no            ckd  
237  ...   40   9600  NaN  yes  yes   no  poor  yes   no            ckd  
..   ...  ...    ...  ...  ...  ...  ...   ...  ...  ...            ...  
117  ...   37   9800  4.4   no   no   no  good   no   no            ckd  
169  ...   27    NaN  NaN  yes  yes   no  good   no  yes            ckd  
191  ...   26   9200  3.4  yes  yes   no  poor   no   no            ckd  
203  ...  NaN    NaN  NaN  yes  yes   no  good   no  yes            ckd  
268  ...   53   8500  4.9   no   no   no  good   no   no         notckd  

[400 rows x 26 columns]

In [ ]:
kidney.rename(columns = {
   "id": "Id",
   "age": "Age",
   "bp": "blood_pressure",
}, inplace=True)

In [ ]:
kidney.head()

Id   Age  blood_pressure     sg   al   su     rbc        pc         pcc  \
0   0  48.0            80.0  1.020  1.0  0.0     NaN    normal  notpresent   
1   1   7.0            50.0  1.020  4.0  0.0     NaN    normal  notpresent   
2   2  62.0            80.0  1.010  2.0  3.0  normal    normal  notpresent   
3   3  48.0            70.0  1.005  4.0  0.0  normal  abnormal     present   
4   4  51.0            80.0  1.010  2.0  0.0  normal    normal  notpresent   

           ba  ...  pcv    wc   rc  htn   dm  cad appet   pe  ane  \
0  notpresent  ...   44  7800  5.2  yes  yes   no  good   no   no   
1  notpresent  ...   38  6000  NaN   no   no   no  good   no   no   
2  notpresent  ...   31  7500  NaN   no  yes   no  poor   no  yes   
3  notpresent  ...   32  6700  3.9  yes   no   no  poor  yes  yes   
4  notpresent  ...   35  7300  4.6   no   no   no  good   no   no   

  classification  
0            ckd  
1            ckd  
2            ckd  
3            ckd  
4            ckd  

[5 rows x 26 columns]

In [ ]:
kidney.isnull()

Id    Age  blood_pressure     sg     al     su    rbc     pc    pcc  \
0    False  False           False  False  False  False   True  False  False   
1    False  False           False  False  False  False   True  False  False   
2    False  False           False  False  False  False  False  False  False   
3    False  False           False  False  False  False  False  False  False   
4    False  False           False  False  False  False  False  False  False   
..     ...    ...             ...    ...    ...    ...    ...    ...    ...   
395  False  False           False  False  False  False  False  False  False   
396  False  False           False  False  False  False  False  False  False   
397  False  False           False  False  False  False  False  False  False   
398  False  False           False  False  False  False  False  False  False   
399  False  False           False  False  False  False  False  False  False   

        ba  ...    pcv     wc     rc    htn     dm    cad  appet     pe  \
0    False  ...  False  False  False  False  False  False  False  False   
1    False  ...  False  False   True  False  False  False  False  False   
2    False  ...  False  False   True  False  False  False  False  False   
3    False  ...  False  False  False  False  False  False  False  False   
4    False  ...  False  False  False  False  False  False  False  False   
..     ...  ...    ...    ...    ...    ...    ...    ...    ...    ...   
395  False  ...  False  False  False  False  False  False  False  False   
396  False  ...  False  False  False  False  False  False  False  False   
397  False  ...  False  False  False  False  False  False  False  False   
398  False  ...  False  False  False  False  False  False  False  False   
399  False  ...  False  False  False  False  False  False  False  False   

       ane  classification  
0    False           False  
1    False           False  
2    False           False  
3    False           False  
4    False           False  
..     ...             ...  
395  False           False  
396  False           False  
397  False           False  
398  False           False  
399  False           False  

[400 rows x 26 columns]

In [ ]:
kidney.isnull().sum()

Id                  0
Age                 9
blood_pressure     12
sg                 47
al                 46
su                 49
rbc               152
pc                 65
pcc                 4
ba                  4
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

In [ ]:
kidney["blood_pressure"].mean()

76.46907216494846

In [ ]:
kidney["blood_pressure"].median()

80.0

In [ ]:
kidney["blood_pressure"].fillna(kidney["blood_pressure"].median(), inplace = True)

In [ ]:
kidney.describe(include = 'all')

Id         Age  blood_pressure          sg          al  \
count   400.000000  391.000000      400.000000  353.000000  354.000000   
unique         NaN         NaN             NaN         NaN         NaN   
top            NaN         NaN             NaN         NaN         NaN   
freq           NaN         NaN             NaN         NaN         NaN   
mean    199.500000   51.483376       76.575000    1.017408    1.016949   
std     115.614301   17.169714       13.489785    0.005717    1.352679   
min       0.000000    2.000000       50.000000    1.005000    0.000000   
25%      99.750000   42.000000       70.000000    1.010000    0.000000   
50%     199.500000   55.000000       80.000000    1.020000    0.000000   
75%     299.250000   64.500000       80.000000    1.020000    2.000000   
max     399.000000   90.000000      180.000000    1.025000    5.000000   

                su     rbc      pc         pcc          ba  ...  pcv    wc  \
count   351.000000     248     335         396         396  ...  330   295   
unique         NaN       2       2           2           2  ...   44    92   
top            NaN  normal  normal  notpresent  notpresent  ...   41  9800   
freq           NaN     201     259         354         374  ...   21    11   
mean      0.450142     NaN     NaN         NaN         NaN  ...  NaN   NaN   
std       1.099191     NaN     NaN         NaN         NaN  ...  NaN   NaN   
min       0.000000     NaN     NaN         NaN         NaN  ...  NaN   NaN   
25%       0.000000     NaN     NaN         NaN         NaN  ...  NaN   NaN   
50%       0.000000     NaN     NaN         NaN         NaN  ...  NaN   NaN   
75%       0.000000     NaN     NaN         NaN         NaN  ...  NaN   NaN   
max       5.000000     NaN     NaN         NaN         NaN  ...  NaN   NaN   

         rc  htn   dm  cad appet   pe  ane classification  
count   270  398  398  398   399  399  399            400  
unique   49    2    5    3     2    2    2              3  
top     5.2   no   no   no  good   no   no            ckd  
freq     18  251  258  362   317  323  339            248  
mean    NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  
std     NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  
min     NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  
25%     NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  
50%     NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  
75%     NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  
max     NaN  NaN  NaN  NaN   NaN  NaN  NaN            NaN  

[11 rows x 26 columns]

In [ ]:
kidney["cad"].fillna(kidney["cad"].mean(), inplace = True)

TypeError: ignored